shared/js/background/utils.es6.js

```diff
@@ -114,6 +138,15 @@ function getAsyncBlockingSupport () {
     return false
 }
 
+// return true if the given url is in the safelist. For checking if the current tab is in the safelist,
+// tabManager.site.whitelisted is the preferred method.
+function isSafeListed (url) {
+    const hostname = extractHostFromURL(url)
+    const tld = tldts.parse(url).domain
+    const safeList = settings.getSetting('whitelisted')
+    return safeList && (safeList[hostname] || safeList[tld])
```

Comment:
should we also check stuff in between hostname and tld? like for a.b.c.example.com hostname is a.b.c.example.com and tld is example.com, but I think c.example.com can be safelisted?